### PR TITLE
316 issue process attribute types by their full names

### PR DESCRIPTION
**Pull request by pfggs**

### CODE REVIEW

File: docs/schemas.md [hunk 1]
```diff
@@ -176,11 +176,3 @@ The same applies when there is more than one level. The last child in the last l
...
 
 > Within JSON Schema, a maximum of **2 nesting levels** is allowed.
-
-## Protobuf Schema
-
-### Not nested messages
-
-As the [Protobuf specification for nested types](https://developers.google.com/protocol-buffers/docs/proto3#nested) says, you can use message types inside other message 
-types, whether they are nested or not. KLoadGen supports it, but you have to define the messages in the same `.proto` file and add a `package` specifier to the `.proto` file on a 
-mandatory basis.
```
Comment (dianaGomezGar) [inline docs/schemas.md:186]: Maybe you should explain how it will work with the new changes or what improvements have you added.

**Review by dianaGomezGar — CHANGES_REQUESTED** [review]

File: src/main/java/net/coru/kloadgen/processor/util/SchemaProcessorUtils.java [hunk 2]
```diff
@@ -223,64 +234,130 @@ private static String checkIfChoppable(final String field) {
...
-  private static void addDefinition(final MessageDefinition.Builder msgDef, final String typeName, final TypeElement typeElement, final HashMap<String, TypeElement> nestedTypes,
-                                    final String packageName, final HashMap<String, TypeElement> rootNestedTypes) {
+  @NotNull
+  private static HashMap<String, TypeElement> processLevelTypes(
+      final HashMap<Integer, HashMap<String, HashMap<String, TypeElement>>> globalNestedTypes, final Builder msgDef, final List<FieldElement> fieldElementList, final int deepLevel,
+      final String messageName) {
+
+    /*
+    We store in a list, all the types instantiated by the attributes.
+     */
```
[the remark this answers is not in the document]
Comment (pfggs) [reply src/main/java/net/coru/kloadgen/processor/util/SchemaProcessorUtils.java]: fixed in new commit

File: src/main/java/net/coru/kloadgen/processor/util/SchemaProcessorUtils.java [hunk 3]
```diff
@@ -223,64 +234,130 @@ private static String checkIfChoppable(final String field) {
...
       } else {
-        msgDef.addField(OPTIONAL, msgDefFieldType, elementField.getName(), elementField.getTag());
+        msgDef.addField(OPTIONAL, elementField.getType(), elementField.getName(), elementField.getTag());
       }
     }
   }
 
-  private static void addDefinition(final MessageDefinition.Builder msgDef, final String typeName, final TypeElement typeElement, final HashMap<String, TypeElement> nestedTypes,
-                                    final String packageName, final HashMap<String, TypeElement> rootNestedTypes) {
+  @NotNull
```
[the remark this answers is not in the document]
Comment (pfggs) [reply src/main/java/net/coru/kloadgen/processor/util/SchemaProcessorUtils.java]: fixed in new commit

File: src/main/java/net/coru/kloadgen/processor/util/SchemaProcessorUtils.java [hunk 4]
```diff
@@ -223,64 +230,119 @@ private static String checkIfChoppable(final String field) {
...
   }
 
-  private static MessageDefinition buildProtoMessageDefinition(final String fieldName, final TypeElement messageElement, final HashMap<String, TypeElement> nestedTypes,
-                                                               final String packageName, final HashMap<String, TypeElement> rootNestedTypes) {
-    fillNestedTypes(messageElement, nestedTypes);
+  private static MessageDefinition buildProtoMessageDefinition(
+      final String fieldName, final TypeElement messageElement, final HashMap<Integer, HashMap<String, HashMap<String, TypeElement>>> globalNestedTypesByLevelAndMessage, int deepLevel) {
+
+    deepLevel++;
+    log.info("Processing deep level {} with fieldName {}...", deepLevel, fieldName);
```
Comment (jemacineiras) [inline src/main/java/net/coru/kloadgen/processor/util/SchemaProcessorUtils.java]: Do we still need this in production?

Comment (pfggs) [reply src/main/java/net/coru/kloadgen/processor/util/SchemaProcessorUtils.java]: No

File: docs/schemas.md (hunk 1 above)
Comment (pfggs) [reply docs/schemas.md:186]: It is not about any added functionality, or new casuistry. This is the correction of a bug in the processing of Protobuf Schemas.

**Review by jemacineiras — APPROVED** [review]

**Pull request merged**